core/server/api/v2/utils/serializers/output/posts.js

```diff
@@ -0,0 +1,31 @@
+const debug = require('ghost-ignition').debug('api:v2:utils:serializers:output:posts');
+const urlService = require('../../../../../services/url');
+
+const urls = (id, attrs, options) => {
+    if (!options.columns || options.columns.includes('url')) {
+        attrs.url = urlService.getUrlByResourceId(id, {absolute: true});
```

Comment:
@gargol Regarding #9962. After `toJSON` the id property is undefined if you use e.g. `fields=url`. I already adapted the change here for v2. v0.1 still needs fixing.